torch/_inductor/fx_passes/reinplace.py

```diff
@@ -460,20 +462,36 @@ def is_meta_only_user(node):
                 return True
         return False
 
+    def _overlap(ls) -> bool:
+        try:
+            return len(compute_overlapping_tensors(ls)) != 0
+        except GuardOnDataDependentSymNode:
+            # If we fail with data dependent error we assume they all overlap.
+            return True
+
     def can_inplace(node, mutated_arg):
         if isinstance(mutated_arg, (list, tuple)):
+            # TODO Using _overlap here causes a several issues.
```

Comment:
How did you want to use it here?

Reply:
basically if we have list(arg0, arg1...) 
if those all do not intersect with share storage we can in-place them all.
so that way i did it is
if _overlap([arg for arg in mutated_arg]):